src/Felix/Instances/Setoid/Raw.agda

```diff
@@ -0,0 +1,91 @@
+{-# OPTIONS --safe --without-K  #-}
+
+module Felix.Instances.Setoid.Raw where
+
+open import Data.Product using (_,_; _,′_; curry′; uncurry′; ∃₂; proj₁; proj₂)
+open import Data.Product.Function.NonDependent.Setoid using (<_,_>ₛ; proj₁ₛ; proj₂ₛ)
+open import Data.Sum using ([_,_]; inj₁; inj₂)
+open import Data.Sum.Function.Setoid using ([_,_]ₛ; inj₁ₛ; inj₂ₛ)
+open import Data.Unit.Polymorphic using (tt)
+open import Function using (Func) renaming (_∘_ to _∘ᵈ_)
+open import Function.Construct.Composition as Comp
+open import Function.Construct.Constant as Const
+open import Function.Construct.Identity as Id
+open import Level using (_⊔_; suc)
+open import Relation.Binary using (Setoid)
+
+open import Felix.Equiv using (Equivalent)
+open import Felix.Raw
+open import Felix.Instances.Setoid.Type
+  using (module setoid-instances; _⟶_; cong; _⟨$⟩_) public
+open setoid-instances public
+
+module setoid-raw-instances where instance
+
+  equivalent : ∀ {c ℓ} → Equivalent (c ⊔ ℓ) {obj = Setoid c ℓ} _⟶_
+  equivalent = record
+    { _≈_ = λ {From} {To} f g →
+      let open Setoid To
+      in ∀ x → f ⟨$⟩ x ≈ g ⟨$⟩ x
+    ; equiv = λ {From} {To} →
+      let open Setoid To
+       in record
+         { refl = λ _ → refl
+         ; sym = λ f≈g x → sym (f≈g x)
+         ; trans = λ f≈g g≈h x → trans (f≈g x) (g≈h x)
+         }
+    }
```

Comment:
Looks a lot like `Function.Construct.Setoid`. Perhaps `Felix.Equiv` should provide a helper to assemble an `Equivalent` from a `Setoid`.

Reply:
`Equiv` provides the reverse as `≈setoid`.